pkg/anago/release.go

```diff
@@ -236,8 +281,71 @@ func (d *DefaultRelease) PushArtifacts(versions []string) error {
 	return nil
 }
 
-func (d *DefaultRelease) PushGitObjects() error { return nil }
+// PushGitObjects uploads to the remote repository the release's tags and branches.
+// Internally, this function calls the release implementation's PushTags,
+// PushBranches and PushMainBranch methods
+func (d *DefaultRelease) PushGitObjects(parentBranch, releaseBranch string, tagList []string) error {
+	// Build the options for the git object pusher
+	pusherOpts := &release.GitObjectPusherOptions{
+		DryRun: !d.options.NoMock,
+		// MaxRetries: options.maxRetries,
+		RepoPath: gitRoot,
+	}
 
-func (d *DefaultRelease) CreateAnnouncement() error { return nil }
+	if err := d.impl.PushTags(pusherOpts, tagList); err != nil {
+		return errors.Wrap(err, "pushing release tags")
+	}
+
+	// Determine which branches have to be pushed
+	branchList := []string{}
+	if releaseBranch != git.DefaultBranch {
+		branchList = append(branchList, releaseBranch)
+
+		// # Additionally push the parent branch if a branch of branch
+		if parentBranch != git.DefaultBranch {
+			branchList = append(branchList, parentBranch)
+		}
+	}
+	// Call the release imprementation PushBranches method
+	if err := d.impl.PushBranches(pusherOpts, branchList); err != nil {
```

Comment:
Yes, that's exactly how it should look like. Now we have the drawback of initialization of the GitPusher 3 times. We could refactor that into a new (mockable) function `NewGitPusher()`, where we afterwards pass the returned instance around:
```suggestion
	if err := d.impl.PushBranches(pusher, pusherOpts, branchList); err != nil {
```

Reply:
Ready, `NewGitPusher()` returns a reusable object passed between all push-related functions